### PR TITLE
Remove KeePassHttp plugin and qhttp

### LINKED ISSUE #913: Remove qhttp and KeePassHTTP functionality
When I create any entry for a website for a known favicon, clicking "download favicon" reports the error " unable to fetch favicon" in all cases
## Expected Behavior

I expect clicking "download favicon" to correctly pull the icon from the database
## Current Behavior

ERROR: unable to fetch favicon
## Steps to Reproduce (for bugs)

1. Create or edit an existing entry for a website with a known favicon
2. Click icon on the left side
3. Click download favicon on the bottom

## Context

I would have to manually import every favicon with a file instead of having them being pulled from the database
## Debug Info

KeePassXC - 2.20
Revision:  caa49a8ef3ee28ed478192389b21d61107b3b8e0

Libraries:
- Qt 5.9.0
- libgcrypt 1.7.7

Operating system: Windows 10 
CPU architecture: 64 bit
Kernel: Windows NT

Enabled extensions:
- KeePassHTTP
- Auto-Type
- YubiKey
[2017-08-30 14-50-17.zip](https://github.com/keepassxreboot/keepassxc/files/1264715/2017-08-30.14-50-17.zip)

### PR DESCRIPTION
**Pull request by droidmonkey**
## Description

*drumroll please*.... this is the official removal of the KeePassHttp support and the qhttp backend. With the integration of the KeePassXC-Browser and native messaging we no longer need this plugin and it was deprecated in 2.3.0.

Closes #913 

## Motivation and context

HTTP servers in a password manager is not a good idea. Also the KeePassHttp protocol is not very secure.

## How has this been tested?

It still compiles!

## Types of changes

- ✅ Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**
- ✅ My change requires a change to the documentation and I have updated it accordingly.

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -93,9 +93,11 @@ These steps place the compiled KeePassXC binary inside the `./build/src/` direct
 
 	```
 	  -DWITH_XC_AUTOTYPE=[ON|OFF] Enable/Disable Auto-Type (default: ON)
-	  -DWITH_XC_HTTP=[ON|OFF] Enable/Disable KeePassHTTP and custom icon downloads (default: OFF)
 	  -DWITH_XC_YUBIKEY=[ON|OFF] Enable/Disable YubiKey HMAC-SHA1 authentication support (default: OFF)
 	  -DWITH_XC_BROWSER=[ON|OFF] Enable/Disable KeePassXC-Browser extension support (default: OFF)
+      -DWITH_XC_NETWORKING=[ON|OFF] Enable/Disable Networking support (favicon download) (default: OFF)
+
+      -DWITH_XC_ALL=[ON|OFF] Enable/Disable compiling all plugins above (default: OFF)
```
Comment (Throne3d) [inline INSTALL.md]: This seems to have different spacing from the rest of the stuff here – looks like they use a combination of spaces and tabs, whereas this is just spaces?

File: cmake/Findqhttpengine.cmake [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+find_path(QHTTPENGINE_INCLUDE_DIR qhttpengine/qhttpengine_export.h)
+
+find_library(QHTTPENGINE_LIBRARY qhttpengine)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(qhttpengine DEFAULT_MSG QHTTPENGINE_LIBRARY QHTTPENGINE_INCLUDE_DIR)
+
+mark_as_advanced(QHTTPENGINE_LIBRARY QHTTPENGINE_INCLUDE_DIR)
```
Comment (yan12125) [inline cmake/Findqhttpengine.cmake]: Is qhttpengine coming back? ;-)

Comment (droidmonkey) [reply cmake/Findqhttpengine.cmake]: Oh yah great catch haha

**Comment by droidmonkey** [comment]
Removed remaining references.

**Comment by TheZ3ro** [comment]
KeePassHTTP, We will miss you :broken_heart:

**Comment by wgml** [comment]
There is still a *Note about KeePassHTTP* section in README.

**Comment by louib** [comment]
me_gusta

**Comment by yan12125** [comment]
default


(Stolen from https://github.com/mozilla-jetpack/jpm/pull/546#issuecomment-231850910, humor from Mozilla :)

**Comment by louib** [comment]
@droidmonkey I guess [this](https://github.com/keepassxreboot/keepassxc/blob/refactor/remove-http/src/gui/MainWindow.cpp#L382) should be removed or updated.

There's also some references to the http connector in `docs/QUICKSTART.md`

**Comment by TheZ3ro** [comment]
Note: I think this will need a blog post about it or a warning paragraph in the release note, just letting users know that this isn't supported anymore, avoiding thousand of new issues.

**Comment by droidmonkey** [comment]
The giant yellow banner in 2.3.0 shown for three start-ups wasnt enough?

**Comment by TheZ3ro** [comment]
Yes, but now we need to tell users that we actually removed it :D

**Comment by droidmonkey** [comment]
Hah of course, we could detect that the service is enabled in the existing configuration file and issue them the final warning. I'll add that to this PR.

**Comment by TheZ3ro** [comment]
I think just a mention in the blog post or the release post is fine, no need to add into KeePassXC itself

**Comment by droidmonkey** [comment]
Sorry for the delay, I removed the additional HTTP references noted by @louib

**Pull request merged**